packages/kerosene/src/array/product.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns the cartesian product of the source arrays
+ *
+ * Usage:
+ * ```typescript
+ * product([1, 2], [3, 4]); // [[1, 3], [1, 4], [2, 3], [2, 4]]
+ * ```
+ *
+ * @param sources
+ */
+export default function product<T>(...sources: T[][]): T[][] {
```

Comment:
Can you make `T` optional? We wouldn't want to enforce typings on such a generic fn

Reply:
`T` will be inferred from the type of the input `sources`, so providing this isn't required for a consumer.